plugins/infinispan52/src/main/java/org/radargun/service/InfinispanServerService.java

```diff
@@ -89,6 +100,42 @@ public void init() {
       clustered = new InfinispanServerClustered(this);
 
       try {
+         if (serverZip != null) {
+            Path homePath = FileSystems.getDefault().getPath(home);
+            Utils.deleteDirectory(homePath.toFile());
+            Files.createDirectory(homePath);
+
+            ZipInputStream zipInputStream = new ZipInputStream(new FileInputStream(serverZip));
+            ZipEntry nextEntry = null;
+            while ((nextEntry = zipInputStream.getNextEntry()) != null) {
+               String entryOutput = homePath.toString() + "/" + nextEntry.getName();
+               if (nextEntry.isDirectory()) {
+                  Files.createDirectory(FileSystems.getDefault().getPath(entryOutput));
+               } else {
+                  FileOutputStream fout = new FileOutputStream(entryOutput);
+                  byte[] buffer = new byte[4096];
+                  for (int c = zipInputStream.read(buffer); c != -1; c = zipInputStream.read(buffer)) {
+                     fout.write(buffer, 0, c);
+                  }
+                  zipInputStream.closeEntry();
+                  fout.close();
+               }
+            }
+
+            List<Path> homeContents = Files.list(homePath).collect(Collectors.toList());
+            // move the server files 1 directory up (remove the extra directory)
+            if (homeContents.size() == 1) {
+               Path fromDir = homeContents.get(0);
+               for (Path path : Files.list(fromDir).collect(Collectors.toList())) {
+                  Files.move(path, FileSystems.getDefault().getPath(home, path.getFileName().toString()));
+               }
+               Files.delete(fromDir);
+            }
+            // the extraction erases the executable bits
+            Path scriptPath = FileSystems.getDefault().getPath(home, "bin", getStartScriptPrefix() + (windows ? "bat" : "sh"));
+            Set<PosixFilePermission> permissions = PosixFilePermissions.fromString("rwxr-xr-x");
```

Comment:
You should probably check  `getFileAttributeView(path, PosixFileAttributeView.class) != null` before this.